src/Xamarin.Android.Build.Tasks/Tasks/GetAdditionalResourcesFromAssemblies.cs

```diff
@@ -193,6 +194,71 @@ bool IsValidDownload(string file, string sha1)
 			return string.Compare (hash, sha1, StringComparison.InvariantCultureIgnoreCase) == 0;
 		}
 
+		void DoDownload (long totalBytes, long offset, Stream responseStream, Stream outputStream, Action<long, long, int> progressCallback = null)
+		{
+			long readSoFar = offset;
+			byte [] buffer = new byte [8192];
+			int bufferSize = buffer.Length;
+			int nread = 0;
+			float percent, tb = (float)totalBytes, lastPercent = 0;
+			outputStream.Seek (offset, SeekOrigin.Begin);
+
+			do {
+				nread = responseStream.Read (buffer, 0, bufferSize);
+				if (nread <= 0)
```

Comment:
I think it would be clearer if you instead did:

    while ((nread = responseStream.Read (buffer, 0, buffer.Length)) > 0) {
        ...
    }